script/src/syscalls/exec_v2.rs

```diff
@@ -0,0 +1,77 @@
+use crate::syscalls::{
+    Place, Source, EXEC, INDEX_OUT_OF_BOUND, SOURCE_ENTRY_MASK, SOURCE_GROUP_FLAG,
+};
+use crate::types::{DataLocation, DataPieceId, ExecV2Args, Message, VmId};
+use ckb_vm::{
+    registers::{A0, A1, A2, A3, A4, A5, A7},
+    Error as VMError, Register, SupportMachine, Syscalls,
+};
+use std::sync::{Arc, Mutex};
+
+pub struct ExecV2 {
+    id: VmId,
+    message_box: Arc<Mutex<Vec<Message>>>,
+}
+
+impl ExecV2 {
+    pub fn new(id: VmId, message_box: Arc<Mutex<Vec<Message>>>) -> ExecV2 {
+        ExecV2 { id, message_box }
+    }
+}
+
+impl<Mac> Syscalls<Mac> for ExecV2
+where
+    Mac: SupportMachine,
+{
+    fn initialize(&mut self, _machine: &mut Mac) -> Result<(), VMError> {
+        Ok(())
+    }
+
+    fn ecall(&mut self, machine: &mut Mac) -> Result<bool, VMError> {
+        if machine.registers()[A7].to_u64() != EXEC {
+            return Ok(false);
+        }
+        let index = machine.registers()[A0].to_u64();
+        let mut source = machine.registers()[A1].to_u64();
+        let place = machine.registers()[A2].to_u64();
+        // To keep compatible with the old behavior. When Source is wrong, a
```

Comment:
I do want to ask here: if old behavior is not a concern, what a proper solution here should be?

Same question goes for `Place` parsing below

Reply:
I suggest removing this part of the compatibility code.